include/openmc/position.h

```diff
@@ -57,9 +58,7 @@ struct Position {
   }
 
   // Data members
-  double x = 0.;
-  double y = 0.;
-  double z = 0.;
+  union{ double xyz[3]; struct{ double x,y,z = 0.0; }; };
```

Comment:
I'm a little concerned about this union since we'd be relying on technically undefined behavior.  It's hard to imagine that any compiler would make it so that `&(p.y) != &(p.xyz[1])`, but I'd still like to stick within the standards if possible.  Maybe we should just leave `Position` unchanged and then fill a different 'double xyz[3]` local variable to use when calling `ray_fire`.  Do you see noticeable performance drop with that data copying?

Reply:
Fair enough. I was thinking that because these were the same type it wouldn't matter and would make interfacing with DAGMC easier, but if it makes the code more vulnerable I'll remove it.

Reply:
I, as of yet, have no evidence that performing that data copy hurts performance significantly, so until I do I'll use the local var. :)